examples/tutorials/nb_python/material_tutorial.py

```diff
@@ -0,0 +1,198 @@
+    os.environ["IMAGEIO_FFMPEG_EXE"] = "/usr/bin/ffmpeg"
+
+repo = git.Repo(".", search_parent_directories=True)
+dir_path = repo.working_tree_dir
+data_path = os.path.join(dir_path, "data")
+
+
+def place_agent(sim):
+    # place our agent in the scene
+    agent_state = habitat_sim.AgentState()
+    agent_state.position = [5.0, 0.0, 1.0]
+    agent_state.rotation = quat_from_angle_axis(
+        math.radians(70), np.array([0, 1.0, 0])
+    ) * quat_from_angle_axis(math.radians(-20), np.array([1.0, 0, 0]))
+    agent = sim.initialize_agent(0, agent_state)
+    return agent.scene_node.transformation_matrix()
+
+
+def make_configuration():
+    # simulator configuration
+    backend_cfg = habitat_sim.SimulatorConfiguration()
+    backend_cfg.scene.id = "data/scene_datasets/habitat-test-scenes/van-gogh-room.glb"
+    backend_cfg.enable_physics = True
+
+    # agent configuration
+    sensor_cfg = habitat_sim.SensorSpec()
+    sensor_cfg.resolution = [1080, 960]
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = [sensor_cfg]
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+def show_obs(sim):
+    data = sim.get_sensor_observations()["rgba_camera"]
+    plt.figure(figsize=(12, 12))
+    plt.imshow(data, interpolation="nearest")
+    plt.axis("off")
+    plt.show(block=False)
+    plt.pause(1)
+
+
+if __name__ == "__main__":
+
+# %%
+    # set up the scene and agent (camera position)
+
+    cfg = make_configuration()
+    sim = habitat_sim.Simulator(cfg)
+    place_agent(sim)
+
+# %%
+    # place a sphere in the scene
+    obj_templates_mgr = sim.get_object_template_manager()
+
+    sphere_template_id = obj_templates_mgr.load_object_configs(
+        str(os.path.join(data_path, "test_assets/objects/sphere"))
+    )[0]
+
+    sphere_ids = []
+    sphere_ids.append(sim.add_object(sphere_template_id))
+    sim.set_translation([3.7, 0.23, 0.0], sphere_ids[0])
+
+    show_obs(sim)
+
+# %%
+    # change the material for the sphere render asset
+
+    # could also use sim.get_object_template(sphere_template_id) instead of
+    # get_object_initialization_template
+    render_asset_handle = sim.get_object_initialization_template(
+        sphere_ids[0]
+    ).render_asset_handle
+    num_materials = sim.get_num_render_asset_materials(render_asset_handle)
+    assert num_materials == 1
+    material_index = 0
+
+    material = sim.get_render_asset_material(render_asset_handle, material_index)
+
+    # make the specular highlights less shiny (more matte)
+    material.shininess *= 0.1
+
+    # reduce the specular highlights (make them darker)
+    material.specular_color *= 0.3
+
+    # set diffuse to dark green
+    material.diffuse_color = mn.Color4(r=0.2, g=0.4, b=0.2)
+
+    # change material for all existing and future uses of this render asset, including
+    # the existing sphere.
+    sim.set_render_asset_material(render_asset_handle, material_index, material)
```

Comment:
From the earlier PR, there was an idea/suggestion to expose the object you're modifying to python, so instead of calling get_render_asset_material and then set_render_asset_material, you just call get and modify the returned object.

Reply:
I'm not a fan of this. I don't particularly like the idea that every modification to a python PhongMaterialInfo object immediately translates to a call into C++ to modify the underlying C++ object. It seems more efficient to set up all the fields of your PhongMaterialInfo in python and then send them all down to C++ in one call to `set_render_asset_material`.

In addition, I don't see this object-expose paradigm used anywhere else in hab-sim. Meanwhile, setters and getters are common, see `set_transformation`, `get_transformation`, `set_rigid_state`, `get_rigid_state`, `set_light_setup`, `get_light_setup`. I'm sticking with that convention.

Reply:
The way all the attributes templates are handled is that the user gets a copy of what is held in the library.  The various editable quantities are treated as a properties of the python-bound user's copy of the object, and when editing is completed, the user's copy of the template is registered, adding/overwriting the library object.  

Now, these are templates that are used to construct things, not actual production constructions, so this paradigm might not work for materials.

Reply:
Can we use material keys here for this? Would be almost exactly the same API as [Light Setups](https://aihabitat.org/docs/habitat-sim/lighting-setups.html)
Something like:
```
material_key = sim.get_render_asset_material_key(render_asset_handle, material_index)
material = sim.get_material(material_key) #this returns a copy of the material info
#or to create a new material
material = habitat_sim.gfx.PhongMaterialInfo()
#then edit the material properties here before setting
sim.set_material(material, material_key) #override or register a key
sim.set_render_asset_material_key(render_asset_handle, material_index, material_key)
```

This will require changes to how we define material keys for assets from ids:
in [MeshMetaData](https://github.com/facebookresearch/habitat-sim/blob/378c6f4287116ff90aae67dd5a4be12bdf884b2e/src/esp/assets/MeshMetaData.h#L80) and [MeshTransformNode
](https://github.com/facebookresearch/habitat-sim/blob/378c6f4287116ff90aae67dd5a4be12bdf884b2e/src/esp/assets/MeshMetaData.h#L31) in [loadMaterial](https://github.com/facebookresearch/habitat-sim/blob/378c6f4287116ff90aae67dd5a4be12bdf884b2e/src/esp/assets/ResourceManager.cpp#L1122)
in [addComponent](https://github.com/facebookresearch/habitat-sim/blob/378c6f4287116ff90aae67dd5a4be12bdf884b2e/src/esp/assets/ResourceManager.cpp#L1528)
and elsewhere in this pipeline. Seems like all of these could be generic string keys now.

Reply:
Ok, I'll take a stab at this.